libtransmission/session.h

```diff
@@ -777,8 +777,31 @@ struct tr_session
 
     [[nodiscard]] bool addressIsBlocked(tr_address const& addr) const noexcept;
 
+    [[nodiscard]] bool has_ip_protocol(tr_address_type type) const noexcept
+    {
+        TR_ASSERT(type == TR_AF_INET || type == TR_AF_INET6);
+        return global_ip_cache_->has_ip_protocol(type);
+    }
+
     [[nodiscard]] tr_address publicAddress(tr_address_type type) const noexcept;
 
+    [[nodiscard]] std::optional<tr_address> globalIP(tr_address_type type) const noexcept
+    {
+        TR_ASSERT(type == TR_AF_INET || type == TR_AF_INET6);
+        return global_ip_cache_->global_addr(type);
+    }
+
+    bool globalIP(tr_address const& addr) noexcept
```

Comment:
I used `external_address` instead of `external_ip` to be consistent with the naming with the rest of the code.

Reply:
I think the term "global" came from the definitions of the IP protocols. IMO it'd be more universal to follow that instead of BEP. Just use "external" when the BEP is involved in the code.